ffi/diplomat/src/fixed_decimal.rs

```diff
@@ -19,14 +21,69 @@ pub mod ffi {
         pub success: bool,
     }
 
+    /// How to round digits when constructing an ICU4XFixedDecimal from a
+    /// floating point number
+    pub enum ICU4XFixedDecimalRoundingMode {
+        /// Truncate leftover digits
+        Truncate,
+        ///  Round up from 0.5
+        HalfExpand,
+    }
+
     impl ICU4XFixedDecimal {
         /// Construct an [`ICU4XFixedDecimal`] from an integer.
+        ///
         /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html) for more information.
         pub fn create(v: i32) -> Box<ICU4XFixedDecimal> {
             Box::new(ICU4XFixedDecimal(FixedDecimal::from(v)))
         }
 
+        /// Construct an [`ICU4XFixedDecimal`] from an float, with enough digits to recover
+        /// the original floating point in IEEE 754 without needing trailing zeros
+        ///
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/fixed_decimal/decimal/struct.FixedDecimal.html#method.from_f64) for more information.
+        pub fn from_float(f: f64) -> Option<Box<ICU4XFixedDecimal>> {
```

Comment:
Nit: use the word "double" for f64 and "float" for f32

Reply:
Hmm, so this gets to an interesting thing: "double" vs "float" is C/C++/Java terminology, Rust just calls them floats (and JS just calls them "numbers"). I kinda feel like we should use the Rust terminology here for consistency? I'm not really opposed to using double here, I just find it a bit weird. Then again, Rust isn't calling these APIs, C++ is.

Reply:
I prefer to either use C++/Java terminology with "double", or use "f64".  But "float" is really misleading because it means a 32-bit IEEE floating point in language that are going to be calling this function.

Reply:
I'll use f64 then